### PR TITLE
Update multi seq

### PR DESCRIPTION
**Pull request by msultan**
- [x] Implement feature / fix bug
 - [x] updated tests
 - [x] Update changelog

changed multi seq featurizer to be consistent with contact featurizer

### CODE REVIEW

**Comment by msultan** [comment]
@brookehus @cxhernandez either of you wanna take a look at this?

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 1]
```diff
@@ -106,37 +108,47 @@ def __init__(self, alignment=None, contacts='all',
...
-                    can_keep.append(residue_index)
-
-            #sort it because i dont want random bs issues.
-            can_keep = np.sort(can_keep)
-            #get its pairs
-            pairs = [i for i in itertools.combinations(can_keep, 2)]
             #create a custom ContactFeaturizer for this particular sequence.
             if self.soft_min:
-                feat_dict[protein] = ContactFeaturizer(pairs,scheme=self.scheme,
+                feat_dict[protein] = ContactFeaturizer(pair_dict[protein],scheme=self.scheme,
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 2]
```diff
@@ -106,37 +108,47 @@ def __init__(self, alignment=None, contacts='all',
...
                                                        ignore_nonprotein=self.ignore_nonprotein,
                                                        soft_min=self.soft_min,
                                                        soft_min_beta=self.soft_min_beta)
             else:
-                feat_dict[protein] = ContactFeaturizer(pairs,scheme=self.scheme,
+                feat_dict[protein] = ContactFeaturizer(pair_dict[protein],scheme=self.scheme,
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 3]
```diff
@@ -106,37 +108,47 @@ def __init__(self, alignment=None, contacts='all',
...
+        for protein in self.protein_list:
+            pair_dict[protein] = []
+        for contact in self.contacts:
+            if self._valid_contact(contact):
+                seq_ind_i , seq_ind_j = contact
+                for protein in self.protein_list:
+                    inv_map = self.all_inv_mappings[protein]
+                    residue_index_i = inv_map[seq_ind_i]
+                    residue_index_j = inv_map[seq_ind_j]
+                    pair_dict[protein].append([residue_index_i,residue_index_j])
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 4]
```diff
@@ -106,37 +108,47 @@ def __init__(self, alignment=None, contacts='all',
...
+            return True
 
     def _create_feat_dict(self):
         feat_dict = {}
+        pair_dict = {}
+        for protein in self.protein_list:
+            pair_dict[protein] = []
+        for contact in self.contacts:
+            if self._valid_contact(contact):
+                seq_ind_i , seq_ind_j = contact
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 5]
```diff
@@ -84,19 +84,21 @@ def __init__(self, alignment=None, contacts='all',
...
         self.same_residue = same_residue
-        self.wanted_positions = contacts
+        self.contacts = contacts
         if contacts is 'all':
             #use the max length(probably a horrible idea)
-            self.wanted_positions = range(max([len(self.alignment[i])
-                                               for i in self.alignment.keys()]))
+            max_seq_len = max([len(self.alignment[i])
+                                       for i in self.alignment.keys()])
+            self.contacts =[i for i in itertools.combinations(np.arange(max_seq_len),2)
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/featurizer/multiseq_featuizer.py [hunk 6]
```diff
@@ -106,37 +108,47 @@ def __init__(self, alignment=None, contacts='all',
 
         self.feat_dict = self._create_feat_dict()
 
+    def _valid_contact(self, contact):
+        seq_ind_i , seq_ind_j = contact
```
Comment (cxhernandez) [inline msmbuilder/featurizer/multiseq_featuizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 7]
```diff
@@ -75,8 +76,9 @@ def test_common_contacts_featurizer_1():
     #do "alignment "
     alignment["actual"] = met_seq+"-"
     alignment["fake"] = fake_met_eq
-
-    feat = CommonContactFeaturizer(alignment=alignment, contacts='all',
+    max_len=max([len(alignment[i]) for i in alignment.keys()])
+    contacts=[i for i in itertools.combinations(np.arange(max_len),2)]
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 8]
```diff
@@ -87,21 +89,24 @@ def test_common_contacts_featurizer_2():
...
     #fake sequence
     fake_met_eq ='FGGFM'
     alignment={}
     #do "alignment "
     alignment["actual"] = met_seq
     alignment["fake"] = fake_met_eq
-
-    feat = CommonContactFeaturizer(alignment=alignment, contacts='all',
+    max_len=max([len(alignment[i]) for i in alignment.keys()])
+    contacts=[i for i in itertools.combinations(np.arange(max_len),2)]
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 9]
```diff
@@ -114,8 +119,9 @@ def test_common_contacts_featurizer_2():
...
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 10]
```diff
@@ -75,8 +76,9 @@ def test_common_contacts_featurizer_1():
     #do "alignment "
     alignment["actual"] = met_seq+"-"
     alignment["fake"] = fake_met_eq
-
-    feat = CommonContactFeaturizer(alignment=alignment, contacts='all',
+    max_len=max([len(alignment[i]) for i in alignment.keys()])
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 11]
```diff
@@ -87,21 +89,24 @@ def test_common_contacts_featurizer_2():
...
+
     #fake sequence
     fake_met_eq ='FGGFM'
     alignment={}
     #do "alignment "
     alignment["actual"] = met_seq
     alignment["fake"] = fake_met_eq
-
-    feat = CommonContactFeaturizer(alignment=alignment, contacts='all',
+    max_len=max([len(alignment[i]) for i in alignment.keys()])
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

File: msmbuilder/tests/test_featurizer.py [hunk 12]
```diff
@@ -114,8 +119,9 @@ def test_common_contacts_featurizer_2():
...
```
Comment (cxhernandez) [inline msmbuilder/tests/test_featurizer.py]: pep8

**Comment by msultan** [comment]
@cxhernandez good now?

**Pull request merged**